src/giftCards/GiftCardUpdate/GiftCardHistory/GiftCardHistory.tsx

```diff
@@ -0,0 +1,252 @@
+import { Link, Typography } from "@material-ui/core";
+import Form from "@saleor/components/Form";
+import Hr from "@saleor/components/Hr";
+import Skeleton from "@saleor/components/Skeleton";
+import Timeline, {
+  TimelineAddNote,
+  TimelineEvent,
+  TimelineNote
+} from "@saleor/components/Timeline";
+import useNotifier from "@saleor/hooks/useNotifier";
+import { orderPath } from "@saleor/orders/urls";
+import { staffMemberDetailsPath } from "@saleor/staff/urls";
+import { GiftCardEventsEnum } from "@saleor/types/globalTypes";
+import React from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+import { IntlShape } from "react-intl";
+
+import { useGiftCardTimelineNoteAddMutation } from "../mutations";
+import { GIFT_CARD_DETAILS_QUERY } from "../queries";
+import { GiftCardAddNote } from "../types/GiftCardAddNote";
+import { GiftCardDetails_giftCard_events } from "../types/GiftCardDetails";
+import useGiftCardHistoryEvents from "./hooks/useGiftCardHistoryEvents";
+import messages from "./messages";
+import useStyles from "./styles";
+
+interface FormData {
+  message: string;
+}
+
+const getEventMessage = (
+  event: GiftCardDetails_giftCard_events,
+  intl: IntlShape
+) => {
+  const getUserOrApp = () => {
+    if (event.user) {
+      const { firstName, lastName } = event.user;
+      return `${firstName} ${lastName}`;
+    }
+    if (event.app) {
+      return event.app.name;
+    }
+  };
+
+  switch (event.type) {
+    case GiftCardEventsEnum.ACTIVATED:
+      return intl.formatMessage(
+        {
+          defaultMessage: "Gift card was activated by {activatedBy}",
+          description: "gift card history message"
+        },
```

Comment:
Messeges in `getEventMessage` could be also imported from `messages.ts` file

Reply:
Some - yes, but not all of them because some return jsx.

Reply:
Ok, but jsx is used only for variables, messages objects (with `defaultMessage` and `description`) can be defined in separate file